app/lib/api_authentication/by_access_token.rb

```diff
@@ -168,7 +168,7 @@ def enforce(access_token)
           System::ErrorReporting.report_error(error)
         end
 
-        connection.transaction(requires_new: true, &Proc.new)
+        connection.transaction(requires_new: true, &block)
```

Comment:
Also possible to use `yield` in case we are sure block would always be passed, but not sure this is a performance critical path to be worth it
```suggestion
        connection.transaction(requires_new: true) { yield }
```

Reply:
To me the current syntax seems quite clear, so I wouldn't change it.